### PR TITLE
fix(cart): default available shipping methods value to an array

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
Changes the Magento cart transformer to set `available_shipping_methods` to an empty array instead of `null`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/cart/src/drivers/magento/transforms/outputs/cart.service.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ export class DaffMagentoCartTransformer {
         ? cart.shipping_addresses[0].available_shipping_methods.map(method =>
           this.shippingRateTransformer.transform(method)
         )
-        : null
+        : []
```
Comment (damienwebdev) [inline libs/cart/src/drivers/magento/transforms/outputs/cart.service.ts:122]: Needs test coverage for the bugfix

Comment (griest024) [reply libs/cart/src/drivers/magento/transforms/outputs/cart.service.ts:122]: added

**Comment by damienwebdev** [comment]
This PR looks good, Github is currently incorrectly reporting check status.

**Pull request merged**